fix(rust): project enroll

twittner opened the pull request:
Currently `project enroll` establishes a secure channel to the project node instead of the project's authority. This commit extends the `ProjectLookup` data to include the authority access route. `project enroll` will attempt to map a `/project` protocol as given in the input multi-address to the corresponding authority address and establish a secure channel to the authority.

**Example**:

```
ockam project enroll \
    --member ... \
    --to /project/myproject/service/authenticator
```

twittner committed:
fix(rust): project enroll

Currently `project enroll` establishes a secure channel to the project
node instead of the project's authority. This commit extends the
`ProjectLookup` data to include the authority access route. `project
enroll` will attempt to map a `/project` protocol as given in the input
multi-address to the corresponding authority address and establish a
secure channel to the authority.

Example:

  ockam project enroll \
    --member ... \
    --to /project/myproject/service/authenticator

SanjoDeundiak commented on implementations/rust/ockam/ockam_command/src/project/enroll.rs:
```diff
@@ -32,33 +35,81 @@ pub struct EnrollCommand {
+        let mut rpc = RpcBuilder::new(&self.ctx, &self.opts, &self.cmd.node_opts.api_node)
+            .tcp(tcp)
+            .build()?;
+        rpc.request(api::create_secure_channel(addr, None)).await?;
```
Shouldn't we know authority IdentityIdentifier here?

SanjoDeundiak commented on implementations/rust/ockam/ockam_command/src/project/enroll.rs:
```diff
@@ -32,33 +35,81 @@ pub struct EnrollCommand {
+///
+/// If the input address begins with a `/project` protocol we look for the
+/// corresponding authority access route and return it.
+fn authority_addr(input: &MultiAddr, map: &ConfigLookup) -> anyhow::Result<Option<MultiAddr>> {
```
I'll move that later, since it'll be used to get credential

SanjoDeundiak commented on implementations/rust/ockam/ockam_command/src/project/enroll.rs:
```diff
@@ -32,33 +35,81 @@ pub struct EnrollCommand {
+    async fn run(self) -> Result<()> {
+        let tcp = TcpTransport::create(&self.ctx).await?;
+        let map = self.opts.config.get_lookup();
+        let to = if let Some(addr) = authority_addr(&self.cmd.to, &map)? {
```
That is fine for now, but I found it potentially misleading that multiaddr starting with `/project/name` may be resolved to either project node or authority node depending on the situation

SanjoDeundiak replied:
cc @mrinalwadhwa

twittner committed:
feat(rust): derive identity identifier from identity

To configure the allowed identity identifier for the establishment of the
secure channel to the project authority, the project lookup info now
includes the authority identity as well as the identity identifier.

SanjoDeundiak commented on implementations/rust/ockam/ockam_command/src/project/util.rs:
```diff
@@ -217,48 +217,62 @@ pub async fn check_project_readiness<'a>(
+                .as_ref()
+                .context("Missing project authority")?;
+            let a = hex::decode(&**a).context("Invalid project authority")?;
+            let v = Vault::default();
```
I don't like that we need to create a Vault here, but let's not bother with that problem for now

twittner committed:
fix(rust): readiness checks for project include authority
